tools/ci_download_clickhouse

```diff
@@ -7,112 +7,136 @@
 #
 
 set -o pipefail
-set -o xtrace
 set -o errexit
 
 SOURCE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 ARG0="$(basename ${BASH_SOURCE[0]})"
 
+TARGET_DIR="out"
+# Location where intermediate artifacts are downloaded / unpacked.
+DOWNLOAD_DIR="$TARGET_DIR/downloads"
+# Location where the final clickhouse directory should end up.
+DEST_DIR="./$TARGET_DIR/clickhouse"
+
 # If you change this, you must also update the md5sums below
 CIDL_VERSION="v21.7"
 source "$SOURCE_DIR/clickhouse_checksums"
-CIDL_ASSEMBLE_DIR="./clickhouse"
 
 # Download from manually-populated S3 bucket for now
 CIDL_URL_BASE="https://oxide-clickhouse-build.s3.us-west-2.amazonaws.com"
 
 function main
 {
-	#
-	# Process command-line arguments.  We generally don't expect any, but
-	# we allow callers to specify a value to override OSTYPE, just for
-	# testing.
-	#
-	if [[ $# != 0 ]]; then
-		CIDL_OS="$1"
-		shift
-	else
-		CIDL_OS="$OSTYPE"
-	fi
-
-	if [[ $# != 0 ]]; then
-		echo "unexpected arguments" >&2
-		exit 2
-	fi
-
-	# Configure this program
-	configure_os "$CIDL_OS"
-	CIDL_URL="$CIDL_URL_BASE/$CIDL_FILE"
-
-	# Download the file.
-	echo "URL: $CIDL_URL"
-	echo "Local file: $CIDL_FILE"
-	do_download_curl "$CIDL_URL" "$CIDL_FILE" || \
-	    fail "failed to download file"
-
-	# Verify the md5sum.
-	calculated_md5="$($CIDL_MD5FUNC "$CIDL_FILE")" || \
-	    fail "failed to calculate md5sum"
-	if [[ "$calculated_md5" != "$CIDL_MD5" ]]; then
-		fail "md5sum mismatch \
-		    (expected $CIDL_MD5, found $calculated_md5)"
-	fi
-
-	# Unpack the tarball into a local directory
-	do_untar "$CIDL_FILE" "$CIDL_ASSEMBLE_DIR"
-
-	# Run the binary as a sanity-check.
-	"$CIDL_ASSEMBLE_DIR/clickhouse" server --version
+  #
```

Comment:
Overall this looks like a good change.  But the reformatting of this function makes it hard to figure out what actually changed here.